SourceControl/SourceControl.tex

```diff
@@ -0,0 +1,741 @@
+\ifx\wholebook\relax\else
+% --------------------------------------------
+% Lulu:
+\documentclass[a4paper,10pt,twoside]{book}
+\usepackage[
+	papersize={6.13in,9.21in},
+	hmargin={.815in,.815in},
+	vmargin={.98in,.98in},
+	ignoreheadfoot
+]{geometry}
+\input{../common.tex}
+\pagestyle{headings}
+\setboolean{lulu}{true}
+% --------------------------------------------
+% A4:
+%	\documentclass[a4paper,11pt,twoside]{book}
+%	\input{../common.tex}
+%	\usepackage{a4wide}
+% --------------------------------------------
+\graphicspath{{figures/} {../figures/}}
+\begin{document}
+\renewcommand{\nnbb}[2]{} % Disable editorial comments
+\fi
+%=================================================================
+\chapter{Storing and sharing source code}
+\label{cha:VersionControl}
+
+The consequence of programming in a world of live objects rather than with files and a text editor is that you have to do something explicit to export your program from your Smalltalk image.
+The old way of doing this, also supported by all Smalltalk dialects, is by creating a \emph{fileout} or a \emph{change set}, which are essentially encoded text files that can be imported into another system.
+A newer way of doing this in \Squeak is to upload your code to a versioned repository on a server using a tool called \ind{Monticello}.
+Both have their strengths and weaknesses: Monticello makes it easier to track and distribute changes, especially when working in a team, while change sets do not require you to set up a server repository and a package organization.
+\seeindex{change set}{file, filing out}
+\index{file!filing out}
+
+In this chapter, we will look at the tools that \Squeak provides for sharing and storing source code:
+
+\begin{itemize}
+	\item{The \menu{Monticello Browser}} is the starting point for loading code from, and saving code in, \ind{Monticello} packages.
+	\item{Using the \menu{Change Sorter},} you can organize your source code modifications, review the code, and write it to a file.
+	\item{The \menu{File List Browser} allows you to browse the local file system on your machine, including an option to load source code from a file into your image.}
+	\item{Through the \menu{Change List Browser}}, you can recover any code changes that you have made to your image, even if you have not saved your image.
+	\item{\menu{Git Tools}} make it possible to manage your source code using the \ind{Git} version control system from within \Squeak.
+\end{itemize}
+
+%=========================================================
+\section{Packages and Monticello}
+\label{sec:packages_and_monticello}
+
+We gave you a quick overview of \ind{Monticello}, \Squeak's packaging tool, in \secref{Monticello}.
+However, Monticello has many more features than were discussed there.
+Monticello manages \emph{Packages}. Before telling you more about Monticello, we will first explain exactly what a \ind{package} is, as Monticello is centered around this concept.
```

Comment:
Monticello manages \emph{Packages}.
Thus, before telling you more about Monticello, we will first explain exactly what a \ind{package} is.